workers/unity/Packages/io.improbable.gdk.core/Subscriptions/SubscriptionAggregate.cs

```diff
@@ -10,28 +10,27 @@ public class SubscriptionAggregate
         private readonly Dictionary<Type, int> typesToSubscriptionIndexes = new Dictionary<Type, int>();
 
         private ISubscriptionAvailabilityHandler availabilityHandler;
-
         private int subscriptionsSatisfied;
 
-        public SubscriptionAggregate(SubscriptionSystem subscriptionSystem, EntityId entityId,
-            params Type[] typesToSubscribeTo)
+        private bool IsSatisfied => subscriptionsSatisfied == subscriptions.Length;
+
+        public SubscriptionAggregate(IReadOnlyList<Type> types, ISubscription[] subscriptions)
         {
-            subscriptions = new ISubscription[typesToSubscribeTo.Length];
+            Assert.AreEqual(types.Count, subscriptions.Length);
 
-            for (int i = 0; i < typesToSubscribeTo.Length; ++i)
+            this.subscriptions = subscriptions;
+            for (var i = 0; i < types.Count; i++)
             {
-                var subscription = subscriptionSystem.Subscribe(entityId, typesToSubscribeTo[i]);
-                typesToSubscriptionIndexes.Add(typesToSubscribeTo[i], i);
-                subscriptions[i] = subscription;
-
-                subscription.SetAvailabilityHandler(Handler.Pool.Rent(this));
+                var type = types[i];
+                typesToSubscriptionIndexes.Add(type, i);
```

Comment:
```suggestion
                typesToSubscriptionIndexes.Add(types[i], i);
```

and remove previous line as we use `type` only here